### PR TITLE
Emotion 11

### LINKED ISSUE #795: Upgrade to Emotion 11
## Context

Emotion 11 was released some time ago:

> Emotion 11 is a slight evolution over Emotion 10. It focuses mainly on the developer experience, TS types improvements, switches internals to hooks, and to the new version of the parser that we use: Stylis.

[Read the full release post](https://emotion.sh/docs/emotion-11)

## Required work

This is a breaking change that needs to be done in a major release (probably v3). The migration itself should be pretty painless for Circuit UI Web and its users.

## Potential blockers

https://github.com/storybookjs/storybook/pull/13300

### PR DESCRIPTION
**Pull request by robinmetral**
Closes #795 

## Purpose

Upgrade to Emotion 11.

## Approach and changes

- Upgrade the Emotion packages and address the breaking changes (see Emotion's [migration guide](https://emotion.sh/docs/emotion-11))
- Tweak the Storybook Webpack config to make it compatible with Emotion 11 ([example](https://github.com/chakra-ui/chakra-ui/blob/a5abb6f9477d87a1cbc0c2d784e009d2bc8a8c6d/.storybook/main.js))
- Switch all component interfaces to extend from `*HTMLAttributes<HTML*Element>` as this is a more accurate typing than `HTMLProps<HTML*Element>`

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: packages/circuit-ui/components/Anchor/Anchor.tsx [hunk 1]
```diff
@@ -40,8 +46,8 @@ export interface BaseProps extends BodyProps {
    */
   ref?: Ref<any>;
 }
-type LinkElProps = Omit<HTMLProps<HTMLAnchorElement>, 'size' | 'onClick'>;
-type ButtonElProps = Omit<HTMLProps<HTMLButtonElement>, 'size' | 'onClick'>;
+type LinkElProps = Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 'onClick'>;
+type ButtonElProps = Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'onClick'>;
```
Comment (connor-baer) [inline packages/circuit-ui/components/Anchor/Anchor.tsx:50]: What motivated this change? Is it something like https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36505#issuecomment-549394273?

File: .storybook/components/BorderWidth.js [hunk 2]
```diff
@@ -15,8 +15,8 @@
 
 import PropTypes from 'prop-types';
 import styled from '@emotion/styled';
-import { css } from '@emotion/core';
-import { ThemeProvider } from 'emotion-theming';
+import { css } from '@emotion/react';
+import { ThemeProvider } from '@emotion/react';
```
[the remark this answers is not in the document]
Comment (robinmetral) [reply .storybook/components/BorderWidth.js:19]: Yes! Nice catch, not sure why this didn't trigger a linting error :thinking:

File: packages/circuit-ui/components/Anchor/Anchor.tsx (hunk 1 above)
Comment (robinmetral) [reply packages/circuit-ui/components/Anchor/Anchor.tsx:50]: Yes, something like this! TS was unhappy with a few things after the migration around HTML element props, so I dug into the React types.

- `HTMLProps` actually include a lot more than just the valid element attributes, that's why we often had to omit the `size` prop for example, although it's not a valid HTML attribute (on most elements)
- `*HTMLAttributes` extends `HTMLAttributes` if there are any extra attributes available for the `*` element. So for example there will be a `ButtonHTMLAttributes` type, but a `div` will simply use `HTMLAttributes`

If you know/find more info on this I'm happy to revisit it, but as far as I could tell this is the most accurate solution :slightly_smiling_face:

File: .storybook/components/BorderWidth.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (robinmetral) [reply .storybook/components/BorderWidth.js]: Done in 14fd3e5

**Review by connor-baer — APPROVED** [review]
1️⃣1️⃣
🕚

**Pull request merged**